### PR TITLE
Backward client go

### PR DESCRIPTION
**Pull request by chenqianfzh**
For the convenience of reviews, this PR is split into two commits:
1. The change to the codegen template file, named "update type generator template", where only one file is involved. 
2. The files which are auto-updated by codegen based on the template changes. 101 files are involved.

Reviewers can browse through the files changes in the second commit quickly.

### CODE REVIEW

**Comment by chenqianfzh** [comment]
PR for issue https://github.com/futurewei-cloud/arktos/issues/185

File: staging/src/k8s.io/code-generator/cmd/client-gen/generators/generator_for_type.go [hunk 1]
```diff
@@ -566,10 +566,22 @@ var createSubresourceTemplate = `
...
+	$if .tenanted$
+	objectTenant := $.inputType|private$.ObjectMeta.Tenant 
+	if objectTenant == "" {
+		objectTenant = c.te
+	}
+	$end$	
+	$if .namespaced$
+	objectTenant := $.inputType|private$.ObjectMeta.Tenant 
+	if objectTenant == "" {
+		objectTenant = c.te
```
Comment (pdgetrf) [inline staging/src/k8s.io/code-generator/cmd/client-gen/generators/generator_for_type.go:578]: what's c.te in this case?

**Comment by Sindica** [comment]
Suggestion: the description of this review are too vague. It is hard to tell what it does later when there are a lot of changes.

File: staging/src/k8s.io/code-generator/cmd/client-gen/generators/generator_for_type.go (hunk 1 above)
Comment (chenqianfzh) [reply staging/src/k8s.io/code-generator/cmd/client-gen/generators/generator_for_type.go:578]: te is the tenant name, defined in type of $.typePlurals$. Together with the field of ns, it defines the space for search for the object to perform operation.

**Pull request merged**